components/AsyncButton.vue

```diff
@@ -40,6 +42,12 @@ const LABEL = {
     waiting: 'Saving&hellip;',
     success: 'Saved',
     error:   'Error',
+  },
+  enable: {
+    action:  'Enable',
+    waiting: 'Enabling&hellip;',
+    success: 'Enabled',
+    error:   'Error',
```

Comment:
1) AsyncButton already has an input to choose which of the labels from the `LABEL` map will be used:  the `mode` prop...

Adding another input to look at query params (in a component that knows nothing about the route) and all that to support this one special case is over-complicated and under-flexible.  The only change you need in this file is these 5 lines.